services/bettersocial/chat/allAnonChat.js

```diff
@@ -3,9 +3,11 @@ const {Op} = require('sequelize');
 const crypto = require('crypto');
 const BetterSocialConstantListUtils = require('../constantList/utils');
 
-const generate_channel_id_for_anon_chat = (owner, member) => {
+const generate_channel_id_for_anon_chat = (owner, member, context = null) => {
   const hash = crypto.createHash('sha256');
-  hash.update(owner, member);
+  hash.update(owner);
+  hash.update(member);
+  hash.update(context);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `context` parameter is optional with a default value of `null`. However, the `hash.update(context)` function will throw an error if `context` is `null`. You should add a condition to check if `context` is not `null` before updating the hash.
```diff
-   hash.update(context);
+   if (context) {
+       hash.update(context);
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->